docs/src/considerations.md

```diff
@@ -96,3 +96,34 @@ The interface is implemented in [`src/tables.jl`](https://github.com/yeesian/Arc
 * `ArchGDAL.FeatureLayer` meets the criteria for an `AbstractRow`-iterator based on the previous bullet and meeting the criteria for [`Iteration`](https://docs.julialang.org/en/v1/manual/interfaces/#man-interface-iteration) in [`base/iterators.jl`](https://github.com/yeesian/ArchGDAL.jl/blob/a665f3407930b8221269f8949c246db022c3a85c/src/base/iterators.jl#L1-L18).
 * `ArchGDAL.AbstractDataset` might contain multiple layers, and might correspond to multiple tables. The way to construct tables would be to get the layers before forming the corresponding tables.
 
+## Missing and Null Semantics in GDAL
+
+When reading the fields of a feature using `getfield(feature, i)`, ArchGDAL observes the following behavior:
+
+| Field | null    | notnull |
+|-------|---------|---------|
+| set   | missing | value   |
+| unset | nothing | nothing |
```

Comment:
Shouldn't the unset/null case be n/a? Because it cannot happen.

Reply:
> Because it cannot happen.

In the (impossible) event that it does, do we prefer receiving a `nothing` or having an error being raised?

Reply:
Hmm I see that since we have `if !isfieldset(feature, i) return nothing` in the code that it would return `nothing`. But since we know it is not possible, putting `nothing` could be confusing to people, that may wrongly believe they have to check for this condition. Hence I think n/a is better. For the same reason it's not worth checking for this condition in `getfield` to raise an error.

Reply:
> But since we know it is not possible, putting `nothing` could be confusing to people, that may wrongly believe they have to check for this condition.

That makes sense, reflected in https://github.com/yeesian/ArchGDAL.jl/pull/238/commits/a7710c24a4101e10f79d79cc01ef6ae8fb09623e, thanks!